packages/dd-trace/test/llmobs/plugins/bedrockruntime/bedrockruntime.spec.js

```diff
@@ -0,0 +1,284 @@
+'use strict'
+
+const agent = require('../../../plugins/agent')
+const Sampler = require('../../../../src/sampler')
+const { DogStatsDClient } = require('../../../../src/dogstatsd')
+const { NoopExternalLogger } = require('../../../../src/external-logger/src')
+
+const nock = require('nock')
+const { expectedLLMObsLLMSpanEvent, deepEqualWithMockValues, MOCK_ANY } = require('../../util')
+const chai = require('chai')
+const LLMObsAgentProxySpanWriter = require('../../../../src/llmobs/writers/spans/agentProxy')
+
+chai.Assertion.addMethod('deepEqualWithMockValues', deepEqualWithMockValues)
+
+const serviceName = 'bedrock-service-name-test'
+
+const PROVIDER = {
+  AI21: 'AI21',
+  AMAZON: 'AMAZON',
+  ANTHROPIC: 'ANTHROPIC',
+  COHERE: 'COHERE',
+  META: 'META',
+  MISTRAL: 'MISTRAL'
+}
+
+describe('Plugin', () => {
+  describe('aws-sdk (bedrockruntime)', function () {
+    before(() => {
+      process.env.AWS_SECRET_ACCESS_KEY = '0000000000/00000000000000000000000000000'
+      process.env.AWS_ACCESS_KEY_ID = '00000000000000000000'
+    })
+
+    after(() => {
+      delete process.env.AWS_SECRET_ACCESS_KEY
+      delete process.env.AWS_ACCESS_KEY_ID
+    })
+
+    withVersions('aws-sdk', ['@aws-sdk/smithy-client', 'aws-sdk'], '>=3', (version, moduleName) => {
+      let AWS
+      let bedrockRuntimeClient
+
+      const bedrockRuntimeClientName =
+        moduleName === '@aws-sdk/smithy-client' ? '@aws-sdk/client-bedrock-runtime' : 'aws-sdk'
+
+      describe('with configuration', () => {
+        before(() => {
+          sinon.stub(LLMObsAgentProxySpanWriter.prototype, 'append')
+
+          // reduce errors related to too many listeners
+          process.removeAllListeners('beforeExit')
+
+          sinon.stub(DogStatsDClient.prototype, '_add')
+          sinon.stub(NoopExternalLogger.prototype, 'log')
+          sinon.stub(Sampler.prototype, 'isSampled').returns(true)
+
+          LLMObsAgentProxySpanWriter.prototype.append.reset()
+
+          return agent.load('aws-sdk', {}, {
+            llmobs: {
+              mlApp: 'test'
+            }
+          })
+        })
+
+        before(done => {
+          const requireVersion = version === '3.0.0' ? '3.422.0' : '>=3.422.0'
+          AWS = require(`../../../../../../versions/${bedrockRuntimeClientName}@${requireVersion}`).get()
+          bedrockRuntimeClient = new AWS.BedrockRuntimeClient(
+            { endpoint: 'http://127.0.0.1:4566', region: 'us-east-1', ServiceId: serviceName }
+          )
+          done()
+        })
+
+        afterEach(() => {
+          nock.cleanAll()
+          LLMObsAgentProxySpanWriter.prototype.append.reset()
+        })
+
+        after(() => {
+          sinon.restore()
+          return agent.close({ ritmReset: false, wipe: true })
+        })
+
+        const prompt = 'What is the capital of France?'
+        const temperature = 0.5
+        const topP = 1
+        const topK = 1
+        const maxTokens = 512
+
+        const models = [
```

Comment:
i liked your idea the other week of maybe moving these fixtures to a shared file - I think `packages/datadog-plugin-aws-sdk/test/fixtures/bedrockruntime.js` might be a good spot